pandas/util/_decorators.py

```diff
@@ -1,21 +1,35 @@
 from functools import wraps
 import inspect
 from textwrap import dedent
-from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
 import warnings
 
 from pandas._libs.properties import cache_readonly  # noqa
 
+FuncType = Callable[..., Any]
+F = TypeVar("F", bound=FuncType)
```

Comment:
Is the bound required here? I thought only had to do with class inheritance so not sure if there's actually something for the Callable or if this is an undocumented feature

Reply:
this is to declare a decorator that preserves the signature of the function it decorates, regardless of that signature. https://mypy.readthedocs.io/en/latest/generics.html#declaring-decorators